src/Fixture/AbstractExampleFactoryAwareFixture.php

```diff
@@ -0,0 +1,146 @@
+    {
+        $options = $this->optionsResolver->resolve($options);
+
+        $i = 0;
+        foreach ($options['custom'] as $resourceOptions) {
+            $resolvedOptions = $this->getExampleFactory()->resolveOptions($resourceOptions);
+            $resource = $this->getExampleFactory()->create($resolvedOptions);
+
+            $this->entityManager->persist($resource);
+
+            ++$i;
+
+            if (0 === ($i % 10)) {
+                $this->entityManager->flush();
+                $this->entityManager->clear();
+            }
+        }
+
+        $this->entityManager->flush();
+        $this->entityManager->clear();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    final public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder($this->getName());
+
+        /** @var ArrayNodeDefinition $optionsNode */
+        $optionsNode = $treeBuilder->getRootNode();
+
+        $nodeBuilder = $optionsNode->children();
+        $nodeBuilder->integerNode('random')->min(0)->defaultValue(0)->end();
+        $nodeBuilder->variableNode('prototype')->end();
+
+        /** @var ArrayNodeDefinition $resourcesNode */
+        $resourcesNode = $optionsNode->children()->arrayNode('custom');
+
+        /** @var ArrayNodeDefinition $resourceNode */
+        $resourceNode = $resourcesNode->requiresAtLeastOneElement()->arrayPrototype();
+        $this->configureResourceNode($resourceNode);
+
+        return $treeBuilder;
+    }
+
+    protected function getExampleFactory(): ExampleFactoryInterface
+    {
+        if (null === $this->exampleFactory) {
+            $exampleFactory = $this->createExampleFactory();
+
+            if (null === $exampleFactory) {
+                throw new \LogicException(sprintf('No example Factory class was found. Use %s or override getExampleFactoryClass method', $this->getExampleFactoryClass()));
+            }
+
+            if (!$exampleFactory instanceof ExampleFactoryInterface) {
+                throw new \LogicException(sprintf('Example Factory "%s" must implement %s.', \get_class($exampleFactory), ExampleFactoryInterface::class));
+            }
+
+            $this->exampleFactory = $exampleFactory;
+        }
+
+        return $this->exampleFactory;
+    }
+
+    protected function getExampleFactoryClass(): string
+    {
+        $names = explode('\\', static::class);
+        $name = end($names);
+        $basename = preg_replace('/(.+)Fixture$/', '${1}', $name);
+
+        $pos = strrpos(static::class, '\\');
+        $namespace = false === $pos ? '' : substr(static::class, 0, $pos);
+
+        return $namespace . '\\Factory\\' . $basename . 'ExampleFactory';
+    }
+
+    protected function configureResourceNode(ArrayNodeDefinition $resourceNode): void
+    {
+        // empty
+    }
+
+    private function createExampleFactory(): ?object
+    {
+        $class = $this->getExampleFactoryClass();
+
+        return class_exists($class) ? new $class() : null;
+    }
```

Comment:
How strongly are bounded to this solution? I'm not strongly against this, but my preferences are to keep away from putting too much logic (especially about dependencies wiring up) from class and inheritance as an extending option. This class will be a service anyway, so I would prefer to go with the autowiring/autoconfiguration way to make it easily accessible for new commers. For now, I would recommend the removal of this semi-magical dependency injection(creation?). 

However, if there will be a willingness in merging the current solution, the problem that I have (despite what I've written above) is that we have only one(pretty neat, I must say 🎉 ) test for this behaviour. My recommendations would be to test `getExampleFactoryClass` should be tested in separation, write additional cases for lack of factories at all, lack of interface in class and for the case, where we are injecting separate factory service.

Reply:
Yes I think you're right. But I'm searching for a solution to have all autoconfigured without any service configuration to do for the developer.

Reply:
I will think about that next week

Reply:
@lchrusciel It was inspired by https://symfony.com/doc/current/validation/custom_constraint.html#creating-the-validator-itself.
I added some examples but maybe this is not a good idea like this...

Reply:
I've just think about this, you often have a resource with just simple properties and faker is enough to make your example factory. In other cases, you just need the entity manager and no other services (using getRepository($className) method. On Sylius world, that's not possible cause you need to use factories and repositories, but in a Symfony project, I think the entity manager is enough maybe all the time.

Reply:
But for Sylius case, the current way still work anyway (via the constructor). On Monofony projects, I override the constructor of the fixture service and specify typehint with the right example factory to use autowiring system.